src/util.ts

```diff
@@ -844,3 +844,45 @@ export function makeMemoizedByteOffsetConverter(buffer: Buffer): (byteOffset: nu
 	};
 }
 
+export class TempFileProvider {
+	private constructor() {}
+	private static _instance: TempFileProvider;
+
+	static get instance() {
+		if (!TempFileProvider._instance) {
+			TempFileProvider._instance = new TempFileProvider();
+		}
+		return TempFileProvider._instance;
+	}
+
+	private store: vscode.Memento;
+
+	/**
+	 * register store to provider for persistance
+	 */
+	registerStore(store: vscode.Memento) {
+		this.store = store;
+	}
+
+	get dir(): string {
```

Comment:
Since `getFilePath` is the only place this is used, can we merge this code into `getFilePath`?